tv_apps/smart-home/js/home.js

```diff
@@ -187,12 +193,77 @@
           cardManager: that.cardManager
         });
         that._cardPicker.on('hide', that.onCardPickerHide.bind(that));
+        that._cardPicker.on('show', that.onCardPickerShow.bind(that));
+      });
+    },
+
+    initContextMenu: function () {
+      this.moveMenuItem.addEventListener('click', () => {
+          this.edit.toggleEditMode();
+          this.edit.once('arranged', () => this.edit.toggleEditMode());
+          this.edit.toggleArrangeMode();
+      });
+
+      this.renameMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
```

Comment:
I think calling rename doesn't depend on mode changing, so you just need one line.